### PR TITLE
[plugin-mobile-app] Adjust element position on screen after swipe to it

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java [hunk 1]
```diff
@@ -139,8 +145,34 @@ public void swipeToElement(SwipeDirection direction, Locator locator, Duration d
...
-        baseValidations.assertElementsNumber(String.format("The element by locator %s exists", locator), elements,
-                ComparisonRule.EQUAL_TO, 1);
+        if (baseValidations.assertElementsNumber(String.format("The element by locator %s exists", locator), elements,
+                ComparisonRule.EQUAL_TO, 1))
+        {
+            adjustElementPosition(elements.get(0), duration);
+        }
+    }
+
+    private void adjustElementPosition(WebElement element, Duration duration)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:155]: `adjustVerticalPosition `

Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:155]: `duration` -> `swipeDuration`

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java [hunk 2]
```diff
@@ -139,8 +145,34 @@ public void swipeToElement(SwipeDirection direction, Locator locator, Duration d
...
+        int visibilityIndent = (int) (VISIBILITY_INDENT_COEFFICIENT * windowSizeHeight);
+        int elementTopCoordinateY = element.getLocation().getY();
+
+        if (elementTopCoordinateY < visibilityIndent)
+        {
+            performSwipe(windowCenterY, visibilityIndent - elementTopCoordinateY, duration);
+        }
+        else if (elementTopCoordinateY > windowSizeHeight - visibilityIndent)
+        {
+            performSwipe(windowCenterY, Math.negateExact(elementTopCoordinateY - (windowSizeHeight - visibilityIndent)),
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:168]: `(windowSizeHeight - visibilityIndent)` - is calculated twice

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java [hunk 3]
```diff
@@ -139,8 +145,34 @@ public void swipeToElement(SwipeDirection direction, Locator locator, Duration d
...
+                    duration);
+        }
+    }
+
+    private void performSwipe(int centerY, int offset, Duration duration)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:173]: what about moving this method to `TouchActions` and naming `performVerticalSwipe`

Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java:173]: `centerY` -> `startY` ?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeCoordinates.java [hunk 4]
```diff
@@ -38,4 +40,25 @@ public Point getEnd()
...
     }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(end, start);
+    }
+
+    @Override
+    public boolean equals(Object obj)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeCoordinates.java:51]: why is it needed?

Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeCoordinates.java:51]: for unit tests :)

Comment (valfirst) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeCoordinates.java:51]: The code for unit tests should live in unit tests

Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeCoordinates.java]: removed as no more needed due api changes

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**